### DISCUSSION TITLE
Run next dev with production environment variables

### DISCUSSION THREAD

**Discussion by JoeRoddy** [vercel/next.js#41461] [Help]
Hi!  I'm trying to start up my development server with my production environment.

I know I can run `next build` and then `npm start` to use prod locally, but I want to make changes and have them live update like `next dev` does.  I'm sure there's a simple way to do this but I'm having trouble finding a solution online.

I assumed it would just be `NODE_ENV=production next dev` but that didn't work for me.

**Comment by icyJoseph** [comment]
Hi,

`NODE_ENV` set to development or production has implications throughout the entire code base because a process `code-elimination` is triggered. All of the code that gives us a good developer experience is stripped when using `NODE_ENV production`. And code that's otherwise verbose is replaced by optimised versions.

Here's an article that goes in depth: https://overreacted.io/how-does-the-development-mode-work/

I think you'd just need to make your production env vars available in `.env.local`, and that should do the trick.

If you application's logic loads variables, or contains logic that depends on `NODE_ENV`, then it'd be better to change that dependency to another ENV var, because you'd be colliding with a "contract" followed by a ton of code in your application, not only by Next.js.
